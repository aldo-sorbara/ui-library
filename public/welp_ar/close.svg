<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
