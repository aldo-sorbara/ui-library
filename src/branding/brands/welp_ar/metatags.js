export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
