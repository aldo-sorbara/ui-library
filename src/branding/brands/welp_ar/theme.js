import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
