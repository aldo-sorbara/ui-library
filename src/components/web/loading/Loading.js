import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { withTheme } from 'styled-components';

import windowSize from 'react-window-size';

class Loading extends React.Component {
  render() {
    const { loading, windowWidth, text, theme } = this.props;
    if (!loading) {
      return null;
    }

    return (
      <Container style={this.props.style} data-w='loading'>
        <div
          style={{
            width: /* istanbul ignore next */ windowWidth > 1000 ? '100%' : 0
          }}
        />
        <LoaderSVG windowWidth={windowWidth}>
          <img src={theme.assetsPaths.loading} alt='Cargando' height='80' />
          {text && (
            <TextContainer>
              <Description>{text}</Description>
            </TextContainer>
          )}
        </LoaderSVG>
      </Container>
    );
  }
}

Loading.propTypes = {
  loading: PropTypes.bool.isRequired,
  text: PropTypes.string
};

Loading.defaultProps = {
  text: null
};

const Container = styled.div`
  display: flex;
  height: 100%;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: ${props => props.theme.zIndex.loaderContainer};
`;

Container.defaultProps = {
  theme: {
    zIndex: {
      loaderContainer: 1020
    }
  }
};

const Description = styled.div`
  font-size: 16px;
  font-weight: bold;
  text-align: center;
`;

const TextContainer = styled.div`
  padding: 48px;
  padding-top: 68px;
`;

const LoaderSVG = styled.div`
  align-items: center;
  background: ${props => props.theme.palette.common.bodyBg};
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: ${/* istanbul ignore next */ props =>
    props.windowWidth >= 1000 ? props.theme.breakPoints.sm : '100%'};
`;

LoaderSVG.defaultProps = {
  theme: {
    palette: {
      common: {
        bodyBg: '#ffffff'
      }
    },
    breakPoints: {
      sm: '768px'
    }
  }
};

export default windowSize(withTheme(Loading));
