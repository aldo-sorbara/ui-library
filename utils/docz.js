export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
