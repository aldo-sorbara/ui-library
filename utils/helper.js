import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
