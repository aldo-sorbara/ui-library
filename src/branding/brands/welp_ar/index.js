import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
