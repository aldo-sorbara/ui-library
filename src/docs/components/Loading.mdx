---
name: Loading
menu: Components
route: /loading
---

import { Playground, PropsTable } from 'docz';
import Loading from '../../components/web/loading';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';

# Emoji

<PropsTable of={Loading} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Loading
        loading
        text='Estamos calculando el mejor prÃ©stamo para vos.'
        style={{ position: 'unset' }}
      />
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/loading--with-default-props)
