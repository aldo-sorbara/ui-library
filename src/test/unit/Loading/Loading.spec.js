import React from 'react';
import Loading from '../../../components/web/loading';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Loading component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Loading loading />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading with custom text', () => {
    const wrapper = mountWithTheme(
      <Loading loading text='Estamos calculando el mejor prÃ©stamo para vos.' />,
      welp_ar.theme
    );
    expect(wrapper.props().text).not.toBe(null);
  });
  it('should render a disabled loading', () => {
    const wrapper = mountWithTheme(<Loading loading={false} />, welp_ar.theme);
    expect(wrapper.props().loading).toBe(false);
  });
  it('should render a big loading', () => {
    const wrapper = mountWithTheme(
      <Loading windowWidth={1024} loading />,
      welp_ar.theme
    );
    expect(wrapper.props().windowWidth > 1000).toBeTruthy();
  });
});
