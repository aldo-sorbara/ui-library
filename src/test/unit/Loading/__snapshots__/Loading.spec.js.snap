// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1020;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="c0"
  data-w="loading"
>
  <div
    style={
      Object {
        "width": 0,
      }
    }
  />
  <div
    className="c1"
  >
    <img
      alt="Cargando"
      height="80"
      src="./public/welp_ar/loader.svg"
    />
  </div>
</div>
`;
